### PR TITLE
gcp - add cloud sql flags resource initial setup

### PR DESCRIPTION
**Pull request by egor-miasnikov**
Added initial classes with `query` and `get` functionality for `Flag` resource in `CloudSQL` service

### CODE REVIEW

File: tools/c7n_gcp/c7n_gcp/resources/sql.py [hunk 1]
```diff
@@ -67,3 +67,14 @@ def get_resource_params(self, model, resource):
         return {'project': project,
                 'instance': instance,
                 'body': {'settings': {'activationPolicy': 'NEVER'}}}
+
+
+@resources.register('sql-flags')
```
Comment (bclements) [inline tools/c7n_gcp/c7n_gcp/resources/sql.py]: Change this to singular ie sql-flag

**Comment by thisisshi** [comment]
i'm sort of confused on the purpose of this resource type, what's the use case for getting the flags?

**Comment by egor-miasnikov** [comment]
> i'm sort of confused on the purpose of this resource type, what's the use case for getting the flags?

it was a requirement from @bclements

File: tools/c7n_gcp/c7n_gcp/resources/sql.py [hunk 2]
```diff
@@ -67,3 +67,14 @@ def get_resource_params(self, model, resource):
...
+
+
+@resources.register('sql-flag')
+class SqlFlags(QueryResourceManager):
+
+    class resource_type(TypeInfo):
+        service = 'sqladmin'
+        version = 'v1beta4'
+        component = 'flags'
+        enum_spec = ('list', "items[]", None)
```
Comment (jondavidi) [inline tools/c7n_gcp/c7n_gcp/resources/sql.py]: try to use consistent quotation marks within a given context. e.g. changing to 'items[]'

File: tools/c7n_gcp/tests/test_sql.py [hunk 3]
```diff
@@ -82,3 +82,14 @@ def test_delete_instance(self):
             self.fail('found deleted instance: %s' % result)
         except HttpError as e:
             self.assertTrue("does not exist" in str(e))
+
+
+class SqlFlagsTest(BaseTest):
```
Comment (jondavidi) [inline tools/c7n_gcp/tests/test_sql.py]: change to singular

File: tools/c7n_gcp/tests/test_sql.py [hunk 4]
```diff
@@ -82,3 +82,14 @@ def test_delete_instance(self):
...
+    def test_sqlflags_query(self):
```
Comment (jondavidi) [inline tools/c7n_gcp/tests/test_sql.py]: change to singular

File: tools/c7n_gcp/c7n_gcp/resources/sql.py [hunk 5]
```diff
@@ -67,3 +67,14 @@ def get_resource_params(self, model, resource):
         return {'project': project,
                 'instance': instance,
                 'body': {'settings': {'activationPolicy': 'NEVER'}}}
+
+
+@resources.register('sql-flag')
+class SqlFlags(QueryResourceManager):
```
[the remark this answers is not in the document]
Comment (egor-miasnikov) [reply tools/c7n_gcp/c7n_gcp/resources/sql.py]: fixed

File: tools/c7n_gcp/c7n_gcp/resources/sql.py (hunk 2 above)
Comment (egor-miasnikov) [reply tools/c7n_gcp/c7n_gcp/resources/sql.py]: fixed

File: tools/c7n_gcp/tests/test_sql.py (hunk 3 above)
Comment (egor-miasnikov) [reply tools/c7n_gcp/tests/test_sql.py]: fixed

File: tools/c7n_gcp/tests/test_sql.py (hunk 4 above)
Comment (egor-miasnikov) [reply tools/c7n_gcp/tests/test_sql.py]: fixed

**Comment by egor-miasnikov** [comment]
> To reduce the test results, I'd change your test to filter for sql databases that are readonly.
> 
> To do this, create a sql instance and set the database to be read only. Then in your test, change the test to filter for instances that have the sql flag readonly set to true. This will allow you to validate that this c7n resource works and it also reduces your test result set.

@bclements this resource just for getting all possible flag for MySQL/Postgres without any specific instances and etc.

**Comment by kapilt** [comment]
the test volume on this seems unavoidable, its a generic response for all options. seems fine to me.

**Comment by egor-miasnikov** [comment]
@bclements @jondavidi is this PR ready for going to `upstream`? If so please click `Approve`

**Comment by jondavidi** [comment]
@bclements will leave approving this to you to make sure your comments above were addressed fully

File: tools/c7n_gcp/c7n_gcp/resources/sql.py (hunk 1 above)
Comment (egor-miasnikov) [reply tools/c7n_gcp/c7n_gcp/resources/sql.py]: also, fixed

**Pull request closed without merging**